applications/fishing-map/src/utils/files.ts

```diff
@@ -0,0 +1,27 @@
+export const blobToText = (blob: Blob): Promise<string> => {
+  return new Promise((resolve, reject) => {
+    const reader = new FileReader()
+    reader.onloadend = () => {
+      if (reader.result) {
+        resolve(reader.result as string)
+      } else {
+        reject('no reader result')
+      }
+    }
+    reader.readAsText(blob)
+  })
+}
+
+export const blobToArrayBuffer = (blob: Blob): Promise<ArrayBuffer> => {
```

Comment:
I didn't know how to merge these two methods supporting different return types @fertandil87 @rdgfuentes any idea?

Reply:
I think you could use [function overloads](https://www.typescriptlang.org/docs/handbook/functions.html#overloads) to accomplish that. @j8seangel . If the blob parameter received is of different types (eg: ZipFile or GeoJsonFile then you can define the two signatures for it and in the fn body return one type or the other depending on the blob argument type.